src/twister2/builder/west_builder.py

```diff
@@ -76,6 +76,10 @@ def _prepare_cmake_args(self, build_config: BuildConfig) -> list[str]:
     def _prepare_args(args: list[str]) -> list[str]:
         return ['-D{}'.format(arg.replace('"', '')) for arg in args]
 
+    @staticmethod
+    def _prepare_extra_configs(args: list[str]) -> list[str]:
```

Comment:
Is there any reason we need another format of args for `extra_configs`?

Reply:
Yes, it is:
We cannot remove `"`  for `extra_configs`. When you want to pass some configs (e.g. `CONFIG_MCUBOOT_IMAGE_VERSION="1.1.1+1"`) via `west` you must keep quotes.
For `extra_args` we should remove quotes (e.g. from OVERLAY_CONFIG).
Discussed with @gopiotr